tests/backup/backup_test.go

```diff
@@ -20,15 +61,642 @@ func TestBackup(t *testing.T) {
+				kubeconfigPath, err))
+		clusterCreateReq := &api.ClusterCreateRequest{
+			CreateMetadata: &api.CreateMetadata{
+				Name:  name,
+				OrgId: orgID,
+			},
+			Cluster: &api.ClusterInfo{
+				Kubeconfig:      base64.StdEncoding.EncodeToString(kubeconfigRaw),
+				CloudCredential: cloudCred,
+			},
+		}
+		_, err = backupDriver.CreateCluster(clusterCreateReq)
+		Expect(err).NotTo(HaveOccurred(),
+			fmt.Sprintf("Failed to create cluster [%s] in org [%s]. Error : [%v]",
+				name, orgID, err))
+	})
+}
+
+// CreateBackup creates backup
+func CreateBackup(backupName string, clusterName string, bLocation string,
+	namespaces []string, labelSelectores map[string]string, orgID string) {
+
+	Step(fmt.Sprintf("Create backup [%s] in org [%s] from cluster [%s]",
+		backupName, orgID, clusterName), func() {
+
+		backupDriver := Inst().Backup
+		bkpCreateRequest := &api.BackupCreateRequest{
+			CreateMetadata: &api.CreateMetadata{
+				Name:  backupName,
+				OrgId: orgID,
+			},
+			BackupLocation: bLocation,
+			Cluster:        clusterName,
+			Namespaces:     namespaces,
+			LabelSelectors: labelSelectores,
+		}
+		_, err := backupDriver.CreateBackup(bkpCreateRequest)
+		Expect(err).NotTo(HaveOccurred(),
+			fmt.Sprintf("Failed to create backup [%s] in org [%s]", backupName, orgID))
+		// TODO: validate createClusterResponse also
+
+	})
+}
+
+// CreateRestore creates restore
+func CreateRestore(restoreName string, backupName string,
+	namespaceMapping map[string]string, clusterName string, orgID string) {
+
+	Step(fmt.Sprintf("Create restore [%s] in org [%s] on cluster [%s]",
+		restoreName, orgID, clusterName), func() {
+
+		backupDriver := Inst().Backup
+		createRestoreReq := &api.RestoreCreateRequest{
+			CreateMetadata: &api.CreateMetadata{
+				Name:  restoreName,
+				OrgId: orgID,
+			},
+			Backup:           backupName,
+			Cluster:          clusterName,
+			NamespaceMapping: namespaceMapping,
+		}
+		_, err := backupDriver.CreateRestore(createRestoreReq)
+		Expect(err).NotTo(HaveOccurred(),
+			fmt.Sprintf("Failed to create restore [%s] in org [%s] on cluster [%s]",
+				restoreName, orgID, clusterName))
+		// TODO: validate createClusterResponse also
+	})
+}
+
+// DeleteBackup deletes backup
+func DeleteBackup(backupName string, clusterName string, orgID string) {
+
+	Step(fmt.Sprintf("Delete backup [%s] in org [%s] from cluster [%s]",
+		backupName, orgID, clusterName), func() {
+
+		backupDriver := Inst().Backup
+		bkpDeleteRequest := &api.BackupDeleteRequest{
+			Name:  backupName,
+			OrgId: orgID,
+		}
+		backupDriver.DeleteBackup(bkpDeleteRequest)
+		// Best effort cleanup, dont fail test, if deletion fails
+		//Expect(err).NotTo(HaveOccurred(),
+		//	fmt.Sprintf("Failed to delete backup [%s] in org [%s]", backupName, orgID))
+		// TODO: validate createClusterResponse also
+	})
+}
+
+func discoverProvider() string {
+	if os.Getenv("AWS_ACCESS_KEY_ID") != "" && os.Getenv("AWS_SECRET_ACCESS_KEY") != "" {
```

Comment:
@stgleb mere presence of AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY  env variables should not be the sole condition for determining whether to create org, cloud-creds, backup-locations etc. in BeforeSuite(). In future, these env vars can be passed to torpedo for different purposes like cloud-drive validations in ASG test cases etc.

So, either we should have an explicit way to tell to torpedo to create px-backup objects or completely remove objects creation from BeforeSuite() and let them get created only when user wants to run backup tests.

Reply:
I agree, as we decided - lets introduce single env variable responsible for determining type of cloud provider the test is running on.